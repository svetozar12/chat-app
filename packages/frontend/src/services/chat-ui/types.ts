import React from 'react';

export interface IBaseComponent<ChakraPropsType> {
  style?: React.CSSProperties;
  baseProps?: React.ComponentPropsWithRef<any>;
  chakraProps?: ChakraPropsType;
}
