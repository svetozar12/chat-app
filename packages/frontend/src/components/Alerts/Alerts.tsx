import React from "react";
import { useSelector, useDispatch } from "react-redux";
import { InitialState } from "redux/state";
import { css } from "@emotion/css";
import { IoClose } from "react-icons/io5";
export const Alerts = () => {
  const dispatch = useDispatch();
  const state = useSelector(
    (state: { authReducer: InitialState }) => state.authReducer,
  );
  if (!state.good && !state.bad)
    return (
      <div
        className={css`
          width: 100%;
          min-height: 3.5rem;
          max-height: 10rem;
          margin-bottom: 1rem;
        `}
      ></div>
    );
  const closeAlert = () => {
    dispatch({ type: "LOGIN_POST_ERROR", bad: "" });
    dispatch({ type: "REGISTER_POST", good: "" });
    dispatch({ type: "REGISTER_POST_ERROR", bad: "" });
  };
  return (
    <div
      className={css`
        position: relative;
        border-radius: 5px;
        background: var(--bad-alert-bgcolor);
        width: 70%;
        padding: 0.2rem 0;
        height: auto;
        min-height: 3.5rem;
        max-height: 10rem;
        display: flex;
        margin-bottom: 1rem;
        align-items: center;
        justify-content: center;
        font-size: 1.9rem;
        color: ${state.good
          ? "var(--good-alert-color)"
          : "var(--bad-alert-color)"};
        background: ${state.good
          ? "var(--good-alert-bgcolor)"
          : "var(--bad-alert-bgcolor)"};
        border-left: 5px solid
          ${state.good ? "var(--good-alert-color)" : "var(--bad-alert-color)"};
      `}
    >
      <p className="flex" style={{ width: "95%", margin: "0" }}>
        {state.good || state.bad}
      </p>
      <div
        className={css`
          display: flex;
          align-items: center;
          justify-content: center;
          height: 112%;
          width: 5%;
          padding: 0 0.5rem;
          border-top-right-radius: 5px;
          border-bottom-right-radius: 5px;
        `}
      >
        <IoClose
          onClick={closeAlert}
          className={css`
            width: 2.5rem;
            height: 2.5rem;
            cursor: pointer;
            &:hover {
              color: var(--main-black);
            }
          `}
        />
      </div>
    </div>
  );
};
