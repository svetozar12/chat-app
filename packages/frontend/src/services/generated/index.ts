export * from './gql';
export * from './ssr';
