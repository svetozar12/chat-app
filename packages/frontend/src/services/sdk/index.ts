// resources
import user from './resources/users';
import auth from './resources/auth';
import message from './resources/message';
import invite from './resources/invite';
import chatroom from './resources/chatroom';
// types
import axios from 'axios';
import { constants } from '../../constants';
const client = axios.create({ baseURL: constants.GRAPHQL_URL, method: 'POST', headers: { 'Content-Type': 'application/json' } });

const sdk = {
  user,
  auth,
  message,
  invite,
  chatroom,
};

export { client };
export default sdk;
