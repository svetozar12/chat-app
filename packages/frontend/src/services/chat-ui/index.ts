import Alerts from 'services/chat-ui/Alerts';
import DefaultLink from 'services/chat-ui/DefaultLink';
import HamburgerMenu from 'services/chat-ui/HamburgerMenu';
import Modal from 'services/chat-ui/Modal';
import RadioCard from 'services/chat-ui/RadioCards/RadioCards';
import SingleAvatar from 'services/chat-ui/SingleAvatar';
// types
import { IBaseComponent } from 'services/chat-ui/types';
import { IAlerts } from 'services/chat-ui/Alerts/Alerts';

export { Alerts, DefaultLink, HamburgerMenu, Modal, RadioCard, SingleAvatar };
export type { IBaseComponent, IAlerts };
