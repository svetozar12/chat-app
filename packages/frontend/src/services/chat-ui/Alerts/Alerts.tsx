import { useEffect } from 'react';
import { Alert, AlertIcon, AlertStatus, AlertTitle, CloseButton, HStack, ScaleFade, StackProps } from '@chakra-ui/react';
import { IBaseComponent } from 'services/chat-ui';
type Base = IBaseComponent<StackProps>;

interface IAlerts extends Base {
  message: string;
  closeAlert: () => void;
  type: AlertStatus;
}

function Alerts(props: IAlerts) {
  const { type, message, closeAlert, chakraProps, baseProps, style } = props;

  useEffect(() => {
    setTimeout(() => {
      closeAlert();
    }, 5000);
  }, []);

  return (
    <HStack pos="absolute" transform="translate(50%,-50%)" top="20%" right="50%" {...chakraProps} {...style} {...baseProps}>
      <Alert status={type} zIndex="99999">
        <AlertIcon w="20" h="20" />
        <AlertTitle textAlign="center" w="full">
          {message}
        </AlertTitle>
        <CloseButton alignSelf="flex-start" position="relative" right={0} top={0} onClick={closeAlert} />
      </Alert>
    </HStack>
  );
}
export type { IAlerts };
export default Alerts;
