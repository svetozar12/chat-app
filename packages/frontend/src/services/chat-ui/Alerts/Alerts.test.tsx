import { mount, ReactWrapper } from 'enzyme';
import Alerts from 'services/chat-ui/Alerts';
import { AlertTitle, Alert, HStack } from '@chakra-ui/react';
import { IAlerts } from 'services/chat-ui';

const componentProps: IAlerts = {
  type: 'success',
  message: 'test message',
  closeAlert: () => console.log('test func'),
  style: { background: 'lime' },
  chakraProps: { h: '20%' },
};

const baseProps = {
  pos: 'absolute',
  w: '60%',
  transform: 'translate(50%,-600%)',
  top: '50%',
  right: '50%',
  h: '20%',
  background: 'lime',
};

const alertTypes: Array<'success' | 'info' | 'warning' | 'error'> = ['success', 'info', 'warning', 'error'];

const setupMount = () => {
  return mount(<Alerts {...componentProps} />);
};

describe(`testing correct behaviour of <Alerts />`, () => {
  let wrapper: ReactWrapper;
  beforeAll(() => {
    wrapper = setupMount();
  });
  alertTypes.forEach((type) => {
    const innerWrapper = mount(<Alerts {...{ ...componentProps, type }} />);
    it(`component <Alerts /> should have prop(status) with value ${type}`, () => {
      expect(innerWrapper.find(Alert).prop('status')).toBe(type);
    });
  });
  it('component <Alerts /> should render', () => {
    expect(wrapper.find(Alerts).length).toBe(1);
  });
  it(`component <Alerts /> should have prop(status) with value ${componentProps.type}`, () => {
    expect(wrapper.find(Alert).prop('status')).toBe(componentProps.type);
  });
  it(`component <Alerts /> should have text with value ${componentProps.message}`, () => {
    expect(wrapper.find(AlertTitle).text()).toBe('test message');
  });
  Object.entries(baseProps).forEach(([key, value]) => {
    it(`component <Alerts /> should have prop ${key}:${value}`, () => {
      expect(wrapper.find(HStack).prop(key)).toBe(value);
    });
  });
  it('component <Alerts /> should match snapshot', () => {
    expect(wrapper).toMatchSnapshot();
  });
});
