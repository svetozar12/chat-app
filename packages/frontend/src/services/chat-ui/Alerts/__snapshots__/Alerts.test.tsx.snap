// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`testing correct behaviour of <Alerts /> component <Alerts /> should match snapshot 1`] = `
<Alerts
  chakraProps={
    Object {
      "h": "20%",
    }
  }
  closeAlert={[Function]}
  message="test message"
  style={
    Object {
      "background": "lime",
    }
  }
  type="success"
>
  <HStack
    background="lime"
    h="20%"
    pos="absolute"
    right="50%"
    top="50%"
    transform="translate(50%,-600%)"
    w="60%"
  >
    <Stack
      align="center"
      background="lime"
      direction="row"
      h="20%"
      pos="absolute"
      right="50%"
      top="50%"
      transform="translate(50%,-600%)"
      w="60%"
    >
      <Styled(div)
        __css={
          Object {
            "& > *:not(style) ~ *:not(style)": Object {
              "marginBottom": 0,
              "marginEnd": 0,
              "marginStart": "0.5rem",
              "marginTop": 0,
            },
          }
        }
        alignItems="center"
        background="lime"
        className="chakra-stack"
        display="flex"
        flexDirection="row"
        h="20%"
        pos="absolute"
        right="50%"
        top="50%"
        transform="translate(50%,-600%)"
        w="60%"
      >
        <Noop />
        <div
          className="chakra-stack css-0"
        >
          <Alert
            key=".0"
            status="success"
          >
            <Styled(div)
              __css={
                Object {
                  "alignItems": "center",
                  "display": "flex",
                  "overflow": "hidden",
                  "position": "relative",
                  "width": "100%",
                }
              }
              className="chakra-alert"
              role="alert"
            >
              <Noop />
              <div
                className="chakra-alert css-0"
                role="alert"
              >
                <AlertIcon>
                  <Styled(span)
                    className="chakra-alert__icon"
                    display="inherit"
                  >
                    <Noop />
                    <span
                      className="chakra-alert__icon css-0"
                    >
                      <CheckIcon
                        h="100%"
                        w="100%"
                      >
                        <Icon
                          h="100%"
                          viewBox="0 0 24 24"
                          w="100%"
                        >
                          <Styled(svg)
                            __css={
                              Object {
                                "color": "currentColor",
                                "display": "inline-block",
                                "flexShrink": 0,
                                "h": "1em",
                                "lineHeight": "1em",
                                "w": "1em",
                              }
                            }
                            className="chakra-icon"
                            focusable={false}
                            h="100%"
                            verticalAlign="middle"
                            viewBox="0 0 24 24"
                            w="100%"
                          >
                            <Noop />
                            <svg
                              className="chakra-icon css-0"
                              focusable={false}
                              viewBox="0 0 24 24"
                            >
                              <path
                                d="M12,0A12,12,0,1,0,24,12,12.014,12.014,0,0,0,12,0Zm6.927,8.2-6.845,9.289a1.011,1.011,0,0,1-1.43.188L5.764,13.769a1,1,0,1,1,1.25-1.562l4.076,3.261,6.227-8.451A1,1,0,1,1,18.927,8.2Z"
                                fill="currentColor"
                              />
                            </svg>
                          </Styled(svg)>
                        </Icon>
                      </CheckIcon>
                    </span>
                  </Styled(span)>
                </AlertIcon>
                <AlertTitle
                  textAlign="center"
                  w="full"
                >
                  <Styled(div)
                    className="chakra-alert__title"
                    textAlign="center"
                    w="full"
                  >
                    <Noop />
                    <div
                      className="chakra-alert__title css-0"
                    >
                      test message
                    </div>
                  </Styled(div)>
                </AlertTitle>
                <CloseButton
                  alignSelf="flex-start"
                  onClick={[Function]}
                  position="relative"
                  right={0}
                  top={0}
                >
                  <Styled(button)
                    __css={
                      Object {
                        "alignItems": "center",
                        "display": "flex",
                        "flexShrink": 0,
                        "justifyContent": "center",
                        "outline": 0,
                      }
                    }
                    alignSelf="flex-start"
                    aria-label="Close"
                    onClick={[Function]}
                    position="relative"
                    right={0}
                    top={0}
                    type="button"
                  >
                    <Noop />
                    <button
                      aria-label="Close"
                      className="css-0"
                      onClick={[Function]}
                      type="button"
                    >
                      <CloseIcon
                        height="1em"
                        width="1em"
                      >
                        <Icon
                          aria-hidden={true}
                          focusable="false"
                          height="1em"
                          width="1em"
                        >
                          <Styled(svg)
                            __css={
                              Object {
                                "color": "currentColor",
                                "display": "inline-block",
                                "flexShrink": 0,
                                "h": "1em",
                                "lineHeight": "1em",
                                "w": "1em",
                              }
                            }
                            aria-hidden={true}
                            className="chakra-icon"
                            focusable="false"
                            height="1em"
                            verticalAlign="middle"
                            viewBox="0 0 24 24"
                            width="1em"
                          >
                            <Noop />
                            <svg
                              aria-hidden={true}
                              className="chakra-icon css-0"
                              focusable="false"
                              viewBox="0 0 24 24"
                            >
                              <path
                                d="M.439,21.44a1.5,1.5,0,0,0,2.122,2.121L11.823,14.3a.25.25,0,0,1,.354,0l9.262,9.263a1.5,1.5,0,1,0,2.122-2.121L14.3,12.177a.25.25,0,0,1,0-.354l9.263-9.262A1.5,1.5,0,0,0,21.439.44L12.177,9.7a.25.25,0,0,1-.354,0L2.561.44A1.5,1.5,0,0,0,.439,2.561L9.7,11.823a.25.25,0,0,1,0,.354Z"
                                fill="currentColor"
                              />
                            </svg>
                          </Styled(svg)>
                        </Icon>
                      </CloseIcon>
                    </button>
                  </Styled(button)>
                </CloseButton>
              </div>
            </Styled(div)>
          </Alert>
        </div>
      </Styled(div)>
    </Stack>
  </HStack>
</Alerts>
`;
